### PR TITLE
482 -- Domain Invitation to You've Been Added To A Domain!

### LINKED ISSUE #482: You have been added to a domain- email notification for .gov accounts
See PB: https://www.google.com/calendar/event?eid=MmUzc3FkNjZxOWl1aWxiODExZTAxNGRydDZfMjAyMzAzMjJUMTgwMDAwWiBsb2dhbi5tY2RvbmFsZEBnc2EuZ292

**Acceptance Criteria**
- Will be sent for users who have get.gov accounts in addition to ones who do not
- Will be sent for users who are added to one or multiple domains (from the domain management page)
-update unit tests
- The subject line and copy is the same for all users (replace old copy)

CC @neilmb 

- [x] [Draft email](https://docs.google.com/document/d/1HnMdniAB9INphrLwKjiACyNLrenyXy0QFf0-yz4EStI/edit?usp=sharing) @michelle-rago 

-------------

Relates to: #507 

implementation note: 
in form_valid on views/domain.py, it may be as simple as always sending an email_invitation.

### PR DESCRIPTION
**Pull request by therealslimhsiehdy**
## Ticket

Resolves #482 

## Changes

- [New subject and wording ](https://docs.google.com/document/d/1HnMdniAB9INphrLwKjiACyNLrenyXy0QFf0-yz4EStI/edit)for the Domain Invitation email

## Context for reviewers

**1.** Once you're logged in, go to the "Domains" section where you have an application in the "Approved" status
Screenshot 2023-09-13 at 4 37 12 PM
**2.** Click "Manage" which is to the right (right of "Approved")
**3.** On the left stacked menu, click "User management"
Screenshot 2023-09-13 at 4 38 37 PM
**4.** From there, click "+ Add another user" and add in your email and then click the "Add user" button
**5.** Check your email!

(See screenshot section for final email)

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [x] Pulled this branch locally and tested it
- [x] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

## Screenshots

**SUBJECT AND TEXT:**

Screenshot 2023-09-21 at 5 09 49 PM

### CODE REVIEW

**Comment by dave-kennedy-ecs** [comment]
LGTM

File: src/registrar/views/domain.py
Comment (rachidatecs) [inline src/registrar/views/domain.py:339]: We want to use DomainInformation or Domain here. Context: DA gets stale after it gets approved and a Domain + DI get created.

Comment (rachidatecs) [reply src/registrar/views/domain.py:339]: I would also move the full_name = domain_information.first.creator here and set "full_name": full_name below. Slightly easier to follow I think.

File: src/registrar/tests/test_views.py
Comment (rachidatecs) [inline src/registrar/tests/test_views.py:1201]: You may still need to create an application and approve it, to get the side effects of domain and domainInfo creation (see my comment on using DI instead of DA in the view file)

File: src/registrar/templates/emails/domain_invitation.txt
Comment (rachidatecs) [inline src/registrar/templates/emails/domain_invitation.txt:1]: Is {% autoescape off %} something we need to implement on all of our txt files?

**Review by rachidatecs — CHANGES_REQUESTED** [review]

File: src/registrar/views/domain.py
Comment (abroddrick) [reply src/registrar/views/domain.py:339]: I will piggy back and agree with the above

File: src/registrar/templates/emails/domain_invitation.txt
Comment (therealslimhsiehdy) [reply src/registrar/templates/emails/domain_invitation.txt:1]: I'm not sure, but just following the pattern of other txt files!

File: src/registrar/templates/emails/domain_invitation.txt
Comment (neilmb) [inline src/registrar/templates/emails/domain_invitation.txt:7]: **comment (non-blocking)**: Because these are plain text emails, we don't have control over the line wrapping that people will see in their email client if we are using these single long lines for each paragraph. I think that it might be good to wrap these text files at something like 60-80 columns wide so that we can ensure shorter lines and correspondingly better comprehension.

Comment (therealslimhsiehdy) [reply src/registrar/templates/emails/domain_invitation.txt]: TIL, thank you for this feedback!!!

**Review by zandercymatics — APPROVED** [review]
LGTM! Great changes... I'd wait for Rachid before merging though, but it seems to be working as expected. Worth noting that for the name it will also return the email w/ first and last name... So "Larry Bob larry.bob@email.com has added you", but I think that is in accordance with the requirements

**Comment by Katherine-Osos** [comment]
@zandercymatics You mentioned the same thing I wanted to bring up. 

@rachidatecs you recently updated how "user" displays so repeat names (e.g. John Smith) would be easy to distinguish in the admin. However, I didn't realize we were using that same field in emails. It's odd to show name/email together like this ... any suggestions?

File: src/registrar/views/domain.py
Comment (rachidatecs) [inline src/registrar/views/domain.py:340]:
```suggestion
           full_name = f"{domaininfo.first().creator.first_name} {domaininfo.first().creator.last_name}"
```
This addresses @zandercymatics and @Katherine-Osos's comments, but check that it passes linter (line seems long)

Comment (rachidatecs) [reply src/registrar/views/domain.py:340]: Approved from me after this change

**Review by rachidatecs — APPROVED** [review]

**Review by Katherine-Osos — APPROVED** [review]

**Pull request merged**
